src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -275,141 +235,38 @@ private void placeBlock(IBlockState localState, @NotNull Block localBlock, @NotN
-
-        return schematic;
-    }
-
-    @NotNull
-    private static File getScanDirectory(@NotNull World world)
-    {
-        File minecolonies;
-        if (world.isRemote)
-        {
-            minecolonies = new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/");
-        }
-        else
-        {
-            minecolonies = MinecraftServer.getServer().getFile("minecolonies/");
-        }
-        checkDirectory(minecolonies);
+        BlockPos blockpos =
+                new BlockPos(Math.min(from.getX(), to.getX()), Math.min(from.getY(), to.getY()), Math.min(from.getZ(), to.getZ()));
+        BlockPos blockpos1 =
+                new BlockPos(Math.max(from.getX(), to.getX()), Math.max(from.getY(), to.getY()), Math.max(from.getZ(), to.getZ()));
+        BlockPos size = blockpos1.subtract(blockpos).add(1, 1, 1);
 
-        @NotNull File scans = new File(minecolonies, "scans/");
-        checkDirectory(scans);
-        return scans;
-    }
+        WorldServer worldserver = (WorldServer) world;
+        MinecraftServer minecraftserver = world.getMinecraftServer();
+        TemplateManager templatemanager = worldserver.getStructureTemplateManager();
 
-    /**
-     * Creates a new tileEntity and formats its data for saving. A new tileEntity is needed to prevent changes to the real one.
-     *
-     * @param schematic  Schematic to add the TileEntity to.
-     * @param tileEntity The tile entity.
-     * @param minPos     The schematic min pos to subtract of the tile entity.
-     */
-    private static void saveTileEntity(@NotNull Schematic schematic, @Nullable TileEntity tileEntity, @NotNull BlockPos minPos)
-    {
-        if (tileEntity != null)
-        {
-            BlockPos newPos = tileEntity.getPos().subtract(minPos);
-            @NotNull NBTTagCompound tileEntityNBT = new NBTTagCompound();
-            tileEntity.writeToNBT(tileEntityNBT);
+        String currentMillis = Long.toString(System.currentTimeMillis());
+        String fileName = "/minecolonies/scans/" + LanguageHandler.format("item.scepterSteel.scanFormat", "", currentMillis + ".nbt");
 
-            TileEntity newTileEntity = TileEntity.createAndLoadEntity(tileEntityNBT);
-            newTileEntity.setPos(newPos);
-            schematic.setTileEntity(newPos, newTileEntity);
-        }
-    }
+        Template template = templatemanager.getTemplate(minecraftserver, new ResourceLocation(fileName));
+        template.takeBlocksFromWorld(world, blockpos, size, true, Blocks.STRUCTURE_VOID);
+        template.setAuthor(Constants.MOD_ID);
 
-    private static void checkDirectory(@NotNull File directory)
-    {
-        if (!directory.exists() && !directory.mkdirs())
-        {
-            Log.getLogger().error("Directory doesn't exist and failed to be created: " + directory.toString());
-        }
+        MineColonies.getNetwork().sendTo(new SaveScanMessage(template.writeToNBT(new NBTTagCompound()), fileName), (EntityPlayerMP) player);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "item.scepterSteel.scanFailure" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)